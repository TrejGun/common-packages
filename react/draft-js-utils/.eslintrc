{
  "extends": [
    "@gemunion/eslint-config/presets/tsx",
    "@gemunion/eslint-config/tests/jest"
  ],
  "parserOptions": {
    "project": "./tsconfig.test.json"
  },
  "root": true
}
