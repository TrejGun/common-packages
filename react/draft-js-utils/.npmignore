/@types
/src
/tsconfig.json
/jest.config.json
/setup-jest.ts
**/*.test.*
