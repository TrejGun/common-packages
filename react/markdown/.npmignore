/src
/@types
/tsconfig.json
/tsconfig.test.json
/*.stories.*
