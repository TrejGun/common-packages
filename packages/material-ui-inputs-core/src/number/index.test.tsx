import React from "react";
import {IntlProvider} from "react-intl";
import {render, cleanup} from "@testing-library/react";
import {MuiThemeProvider, createMuiTheme} from "@material-ui/core";
import {Formik} from "formik";

import {NumberInput} from "./";

afterEach(cleanup);

const i18n = {
  "form.labels.number": "Number",
  "form.placeholders.number": "50",
};

describe("<NumberInput />", () => {
  it("renders positive value", () => {
    const props = {
      name: "number",
      value: "50",
    };

    const formikProps = {
      onSubmit: jest.fn(),
      initialValues: {
        number: "50",
      },
    };

    const {asFragment} = render(
      <MuiThemeProvider theme={createMuiTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <Formik {...formikProps}>
            <NumberInput {...props} />
          </Formik>
        </IntlProvider>
      </MuiThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders negative value", () => {
    const props = {
      name: "number",
      value: "-50",
    };

    const formikProps = {
      onSubmit: jest.fn(),
      initialValues: {
        number: "-50",
      },
    };

    const {asFragment} = render(
      <MuiThemeProvider theme={createMuiTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <Formik {...formikProps}>
            <NumberInput {...props} />
          </Formik>
        </IntlProvider>
      </MuiThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders decimal value", () => {
    const props = {
      name: "number",
      value: "9.99",
    };

    const formikProps = {
      onSubmit: jest.fn(),
      initialValues: {
        number: "9.99",
      },
    };

    const {asFragment} = render(
      <MuiThemeProvider theme={createMuiTheme()}>
        <IntlProvider locale="en" messages={i18n}>
          <Formik {...formikProps}>
            <NumberInput {...props} />
          </Formik>
        </IntlProvider>
      </MuiThemeProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
