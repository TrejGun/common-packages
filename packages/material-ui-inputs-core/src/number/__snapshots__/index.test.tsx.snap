// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput /> renders negative value 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    container="[object HTMLDivElement]"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      Fraction
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        name="data.singleChoiceImageOptions[0].singleChoiceImageOptionFraction"
        placeholder="50"
        type="number"
        value="-50"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NumberInput /> renders positive value 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    container="[object HTMLDivElement]"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      Fraction
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        name="data.singleChoiceImageOptions[0].singleChoiceImageOptionFraction"
        placeholder="50"
        type="number"
        value="50"
      />
    </div>
  </div>
</DocumentFragment>
`;
